NineChronicles.Headless/HostBuilderExtensions.cs

```diff
@@ -87,11 +89,39 @@ RpcNodeServiceProperties properties
                 {
                     hostBuilder.ConfigureKestrel(options =>
                     {
+                        if (properties.HttpOptions is { } httpOptions)
+                        {
+                            options.ListenAnyIP(httpOptions.Port, listenOptions =>
+                            {
+                                listenOptions.Protocols = HttpProtocols.Http1;
+                            });   
+                        }
+
                         options.ListenAnyIP(properties.RpcListenPort, listenOptions =>
                         {
                             listenOptions.Protocols = HttpProtocols.Http2;
                         });
                     });
+
+                    if (properties.HttpOptions is { })
+                    {
+                        hostBuilder.Configure(app =>
+                        {
+                            app.UseRouting();
+
+                            app.UseEndpoints(endpoints =>
+                            {
+                                endpoints.MapMagicOnionHttpGateway("_",
+                                    app.ApplicationServices.GetService<MagicOnion.Server.MagicOnionServiceDefinition>()
+                                        .MethodHandlers, GrpcChannel.ForAddress($"http://{properties.RpcListenHost}:{properties.RpcListenPort}"));
```

Comment:
is it ok without `ChannelCredentials.Insecure` options?

Reply:
Thanks for letting me know it. I also configured `MaxReceiveMessageSize` to unlimited for cases like `GetTip` for genesis block.